web/frontend/src/api/robot.ts

```diff
@@ -20,3 +21,43 @@ export const fetchCurrentOps = (client: Client) => {
     });
   });
 };
+
+export const getResourceNames = (client: Client) => {
+  const request = new robotApi.ResourceNamesRequest();
+
+  return new Promise<ResourceName[]>((resolve, reject) => {
+    client.robotService.resourceNames(request, new grpc.Metadata(), (error, response) => {
+      if (error) {
+        reject(error);
+        return;
+      }
+
+      if (!response) {
+        reject(new Error('An unexpected issue occured.'));
+        return;
+      }
```

Comment:
What would cause this? If it's some sort of internal error, could we add logging or a more detailed error message so we can be more easily alerted?

Otherwise, it feels a little like an unnecessarily suspicious code path that is never called but propagates in the codebase through copy-paste. Could it be replaced with a nullish-coallescing fallback, as in `sensors.ts`?

If we need to keep the check, it may make sense to move this to the same `then` block as data transformation, since it's more about data transformation that the request itself (see comment below)

Reply:
Yeah, this was copy-pasted. Agreed that it's smelly.

>  Could it be replaced with a nullish-coallescing fallback, as in sensors.ts?

Taking this route for now

Reply:
> What would cause this?

No idea. I as well don't like that response can be `null`, since there's no indication to me - the user of the SDK - why that would ever happen.

cc @maximpertsov

Reply:
yes, you should prefer using the `XClient` classes instead of `client.xApi` if they are available. There _should_ be a [deprecation warning](https://ts.viam.dev/modules/robotApi.html) in the docs of each `client.xApi` methods that have a corresponding wrapper - not sure if that comes through in editors.

Reply:
@maximpertsov You can make deprecations more visible to users by adding a @deprecated jsdoc flag to your exports. Here's some usage in a library I maintain: 

https://github.com/michealparks/threlte-xr/blob/main/src/lib/deprecated.ts

Reply:
yeah, [that's what I'm doing](https://github.com/viamrobotics/viam-typescript-sdk/blob/2f66155f51008237d70b226dca5900ae77aa9a35/src/main.ts#L8) to make them appear in the docs.

Reply:
Ah dang! Usually that results in VSCode crossing out that function in my editor but I didn't notice anything 🤔

Reply:
yeah it's weird... it's possible that this doc-comment is treated differently because it's on a re-export vs on the function itself (which is generated code that we don't own). still getting used to the subtleties of js-/ts-doc stuff :sweat_smile:

Reply:
Sidenote: new SDK clients are lookin' nice!